server/graphql/common/comment.ts

```diff
@@ -188,7 +212,7 @@ async function createComment(commentData, req): Promise<Comment> {
     UserId: comment.CreatedByUserId,
     CollectiveId: comment.CollectiveId,
     FromCollectiveId: comment.FromCollectiveId,
-    HostCollectiveId: commentedEntity.collective.approvedAt ? commentedEntity.collective.HostCollectiveId : null,
+    HostCollectiveId: 'HostCollectiveId' in commentedEntity ? commentedEntity.HostCollectiveId : null,
```

Comment:
I don't agree with that part, the `approvedAt` check was there for a reason: we don't want to record the `HostCollectiveId` for other comment types if the collective hasn't been approved yet; otherwise this could be abused to spam a fiscal host's activity log.

And checking the `HostCollectiveId` is not reliable:
- The field doesn't exist for orders
- The field doesn't exist for updates
- For expenses, it's only set at payment time, so we'll be recording comments posted before payment as host-less while comments after payment will have a host attached.

I understand why we want to treat things differently for host applications, but let's preserve the functionality for other comment types:
```
HostCollectiveId:
  commentedEntity instanceof HostApplication ?
    commentedEntity.HostCollectiveId :
    commentedEntity.collective.approvedAt ?
      commentedEntity.collective.HostCollectiveId
      : null